enhancements/network/communication-flows-matrix-ingress.md

```diff
@@ -0,0 +1,282 @@
+
+Periodic tests will be added to `openshift-tests` to validate an up-to-date 
+generated communication matrix matches the communication matrix documented
+in OpenShift release documents.
+
+Another test will be added to validate the ports in a generated communication 
+matrix match a snapshot of the node's listening ports (created with the Linux
+`ss` utility).
+
+The communication matrix module will be used to create and apply firewall rules, 
+run E2E `openshift-tests` and upgrade the OpenShift version to validate the rules 
+are acceptable by the current and next OCP release.
+
+A user will be able to run `openshift-tests` or a new `oc` command, 
+`oc adm communication-matrix validate`, to validate the `EndpointSlices` 
+in the cluster match a current snapshot of the node's listening ports.
+
+### Risks and Mitigations
+
+1. Supporting egress traffic flows (out of scope) is more challenging than ingress as 
+   there is no source of truth object such as the `EndpointSlices` for ingress traffic.
+
+2. For egress flows support an API change might be needed, a change that can affect the 
+   work done for ingress flows and should be reviewed and agreed with the upstream Kubernetes
+   community.
+
+3. Generating the communication matrix on MicroShift and/or in AdminNetworkPolicy format is 
+   not supported yet in the current commatrix module.
+
+### Drawbacks
+N/A
+
+## Open Questions
+
+1. Is there a better OpenShift object to use than `EndpointSlices` to collect 
+   information about ingress traffic?
+
+2. Should a new OpenShift object be introduced to act as the source of truth
+   for cluster traffic?
+
+3. The node hosted services such as sshd, rpc, etc. are currently missing 
+   `EndpointSlices`. We believe it should be created by the Machine API as 
+    part of adding/removing nodes. This can be addressed in later enhancement.
+
+4. Egress traffic support - how to effectively document the expected outgoing
+   traffic from an OpenShift cluster?
+
+5. Is there any additional information that should be added to the communication 
+   matrix (for example the use of the ports)?
+
+## Test Plan
+
+- Unit tests coverage
+
+- E2E tests will be added to `openshift-tests`
+  - Validate an up-to-date generated communication matrix matches the 
+    one documented in OpenShift release documents ([WIP PR](https://github.com/openshift/origin/pull/28946))
+  - Validate the ports in a generated communication matrix match a snapshot
+    of the node's listening ports (created with the Linux `ss` utility) ([WIP PR](https://github.com/openshift/origin/pull/28706))
+  - Use the communication matrix to create and apply firewall rules, and 
+    run E2E `openshift-tests` ([implemented](https://github.com/openshift-kni/commatrix/blob/main/test/e2e/commatrix_suite_test.go#L100))
+  - Use the communication matrix to create and apply firewall rules, and upgrade
+    OpenShift version
```

Comment:
These aren't things that would be "added to `openshift-tests`"; they're both new jobs rather than additions to `openshift-tests`. (So they should be a new top-level bullet point, not under the `openshift-tests` bullet point.)